### PR TITLE
No id swap redux

### PR DESCRIPTION
**Pull request by sethalves**

### CODE REVIEW

File: examples/html/entityProperties.html [hunk 1]
```diff
@@ -358,7 +358,7 @@
                             var types = {};
 
                             for (var i = 0; i < selections.length; i++) {
-                                ids.push(selections[i].id);
+                                ids.push(selections[i].id); // ?
```
Comment (ZappoMan) [inline examples/html/entityProperties.html]: ??

File: examples/libraries/entityList.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ EntityListTool = function(opts) {
 
         var selectedIDs = [];
         for (var i = 0; i < selectionManager.selections.length; i++) {
-            selectedIDs.push(selectionManager.selections[i].id);
+            selectedIDs.push(selectionManager.selections[i].id); // ?
```
Comment (ZappoMan) [inline examples/libraries/entityList.js]: i don't think this will work. we need to test

**Comment by hifi-gustavo** [comment]
Build console output for commit: 782402b4c8f548e1d6d096fb281f680b3717f936 available at: https://gist.github.com/b3b9d6a6abbe20859e8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4920.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4920.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d8beed34d6cf08381e507658a3cf38cdce05f167 available at: https://gist.github.com/b3b9d6a6abbe20859e8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4920.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4920.dmg

File: libraries/entities/src/EntityItemProperties.cpp [hunk 3]
```diff
@@ -863,36 +852,8 @@ bool EntityItemProperties::decodeEntityEditPacket(const unsigned char* data, int
...
-        //newEntityItem._newlyCreated = true;
-        
-        entityID.id = NEW_ENTITY;
-        entityID.creatorTokenID = creatorTokenID;
-        entityID.isKnownID = false;
-
-        valid = true;
-
-        // created time is lastEdited time
-        properties.setCreated(lastEdited);
```
Comment (ZappoMan) [inline libraries/entities/src/EntityItemProperties.cpp]: what about this?

File: libraries/entities/src/MovingEntitiesOperator.h [hunk 4]
```diff
@@ -25,11 +25,11 @@ class EntityToMoveDetails {
 };
 
 inline uint qHash(const EntityToMoveDetails& a, uint seed) {
-    return qHash(a.entity->getEntityItemID(), seed);
+    return qHash(a.entity, seed);
```
Comment (ZappoMan) [inline libraries/entities/src/MovingEntitiesOperator.h]: revert?

File: libraries/entities/src/MovingEntitiesOperator.h [hunk 5]
```diff
@@ -25,11 +25,11 @@ class EntityToMoveDetails {
...
 }
 
 inline bool operator==(const EntityToMoveDetails& a, const EntityToMoveDetails& b) {
-    return a.entity->getEntityItemID() == b.entity->getEntityItemID();
+    return a.entity == b.entity;
```
Comment (ZappoMan) [inline libraries/entities/src/MovingEntitiesOperator.h]: revert?

File: libraries/networking/src/PacketHeaders.h [hunk 6]
```diff
@@ -68,9 +68,9 @@ enum PacketType {
     PacketTypeNodeJsonStats,
     PacketTypeEntityQuery, // 40
     PacketTypeEntityData,
-    PacketTypeEntityAddOrEdit,
+    UNUSED_11,
```
Comment (ZappoMan) [inline libraries/networking/src/PacketHeaders.h]: reuse

File: libraries/networking/src/PacketHeaders.cpp [hunk 7]
```diff
@@ -70,9 +70,10 @@ PacketVersion versionForPacketType(PacketType packetType) {
             return 1;
         case PacketTypeStopNode:
             return 1;
-        case PacketTypeEntityAddOrEdit:
+        case PacketTypeEntityAdd:
+        case PacketTypeEntityEdit:
         case PacketTypeEntityData:
-            return VERSION_ENTITIES_HAVE_FRICTION;
+            return VERSION_NO_ENTITY_ID_SWAP;
```
Comment (ZappoMan) [inline libraries/networking/src/PacketHeaders.cpp]: also add PacketTypeEntityAddResponse (and/or the new packet) to this switch here as well

File: libraries/script-engine/src/ScriptEngine.cpp [hunk 8]
```diff
@@ -321,7 +321,8 @@ void ScriptEngine::init() {
     qScriptRegisterMetaType(this, EntityItemPropertiesToScriptValue, EntityItemPropertiesFromScriptValue);
     qScriptRegisterMetaType(this, EntityItemIDtoScriptValue, EntityItemIDfromScriptValue);
     qScriptRegisterMetaType(this, RayToEntityIntersectionResultToScriptValue, RayToEntityIntersectionResultFromScriptValue);
-    qScriptRegisterSequenceMetaType<QVector<EntityItemID> >(this);
+    qScriptRegisterSequenceMetaType<QVector<QUuid>>(this);
```
Comment (ZappoMan) [inline libraries/script-engine/src/ScriptEngine.cpp]: do you still need this?

**Comment by hifi-gustavo** [comment]
Build console output for commit: aa1d3eb82a020fdf7b5231fc8b7999b1f7d24100 available at: https://gist.github.com/b3b9d6a6abbe20859e8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4920.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4920.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 38edc27cbbf769fb444d8e2f649420072d0e9a75 available at: https://gist.github.com/b3b9d6a6abbe20859e8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4920.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4920.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 52ce26d80d2d1d00f2361fdb046b7d68be22a2d6 available at: https://gist.github.com/b3b9d6a6abbe20859e8c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4920.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4920.dmg

**Pull request merged**